### PR TITLE
README: Add QEMU build instructions for Windows

### PR DESCRIPTION
**Pull request by Filiprogrammer**

### CODE REVIEW

**Review by LecraM2003 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -113,13 +113,130 @@ pip install colorama
...
 ```console
 mkdir qemu_rpi/build
 cd qemu_rpi/build
 ../configure --target-list=aarch64-softmmu --static
 make -j$(nproc)
 ```
 
+For Windows (under Debian):
+
+Install cygwin key and add cygwin sources.
```
Comment (LecraM2003) [inline README.md:127]: Change "." to ":"

File: README.md [hunk 2]
```diff
@@ -113,13 +113,130 @@ pip install colorama
...
+
+```console
+sudo wget https://qemu.weilnetz.de/debian/weilnetz.gpg -P /etc/apt/trusted.gpg.d
+curl -s https://qemu.weilnetz.de/debian/gpg.key | sudo apt-key add -
+echo deb https://qemu.weilnetz.de/debian/ testing contrib | sudo tee /etc/apt/sources.list.d/cygwin.list
+```
+
+Install packages.
```
Comment (LecraM2003) [inline README.md:135]: Change "." to ":"

File: README.md [hunk 3]
```diff
@@ -113,13 +113,130 @@ pip install colorama
...
+```console
+sudo apt-get update
+sudo apt-get install --no-install-recommends \
+  gcc libc6-dev ninja-build \
+  bison flex gettext python3-sphinx \
+  g++-mingw-w64-x86-64 gcc-mingw-w64-x86-64 \
+  mingw-w64-tools mingw64-x86-64-adwaita-icon-theme mingw64-x86-64-cogl mingw64-x86-64-curl mingw64-x86-64-gmp mingw64-x86-64-gnutls mingw64-x86-64-gtk3 mingw64-x86-64-icu mingw64-x86-64-libxml2 mingw64-x86-64-ncurses mingw64-x86-64-sdl2 mingw64-x86-64-usbredir mingw64-x86-64-openssl
+```
+
+Patch gversionmacros.h to work with other versions of glibc.
```
Comment (LecraM2003) [inline README.md:146]: Change "." to ":"

File: README.md [hunk 4]
```diff
@@ -113,13 +113,130 @@ pip install colorama
...
+
+```console
+sed -e '1h;2,$H;$!d;g' -e 's/#if GLIB_VERSION_MIN_REQUIRED > GLIB_VERSION_CUR_STABLE\n#error "GLIB_VERSION_MIN_REQUIRED must be <= GLIB_VERSION_CUR_STABLE"\n#endif\n#if GLIB_VERSION_MAX_ALLOWED < GLIB_VERSION_MIN_REQUIRED\n#error "GLIB_VERSION_MAX_ALLOWED must be >= GLIB_VERSION_MIN_REQUIRED"\n#endif\n#if GLIB_VERSION_MIN_REQUIRED < GLIB_VERSION_2_26\n#error "GLIB_VERSION_MIN_REQUIRED must be >= GLIB_VERSION_2_26"\n#endif//' /usr/x86_64-w64-mingw32/sys-root/mingw/include/glib-2.0/glib/gversionmacros.h > gversionmacros.h_tmp
+sudo mv gversionmacros.h_tmp /usr/x86_64-w64-mingw32/sys-root/mingw/include/glib-2.0/glib/gversionmacros.h
+```
+
+Patch configure script to work with other versions of glibc.
```
Comment (LecraM2003) [inline README.md:153]: Change "." to ":"

File: README.md [hunk 5]
```diff
@@ -113,13 +113,130 @@ pip install colorama
...
+
+```console
+sed 's/^    if $pkg_config --atleast-version=$glib_req_ver $i; then//' configure | sed -e '1h;2,$H;$!d;g' -e 's/    else\n        error_exit "glib-$glib_req_ver $i is required to compile QEMU"\n    fi//' > configure_tmp
+sudo mv configure_tmp configure
+```
+
+Workaround for buggy cross pkg-config
```
Comment (LecraM2003) [inline README.md:160]: Add ":" at the end of the line

File: README.md [hunk 6]
```diff
@@ -113,13 +113,130 @@ pip install colorama
...
+
+```console
+printf '#! /bin/sh\nif [ x"${PKG_CONFIG_LIBDIR+set}" = x ]; then\nbasename="${0##*/}"\ntriplet="${basename%%-pkg-config}"\nmultiarch="`dpkg-architecture -t"${triplet}" -qDEB_HOST_MULTIARCH 2>/dev/null`"\nPKG_CONFIG_LIBDIR="/usr/local/${triplet}/lib/pkgconfig"\nPKG_CONFIG_LIBDIR="$PKG_CONFIG_LIBDIR:/usr/local/share/pkgconfig"\nif [ -n "$multiarch" ]; then\nPKG_CONFIG_LIBDIR="/usr/local/lib/${multiarch}/pkgconfig:$PKG_CONFIG_LIBDIR"\nPKG_CONFIG_LIBDIR="$PKG_CONFIG_LIBDIR:/usr/lib/${multiarch}/pkgconfig"\nfi\nPKG_CONFIG_LIBDIR="$PKG_CONFIG_LIBDIR:/usr/${triplet}/lib/pkgconfig"\nPKG_CONFIG_LIBDIR="$PKG_CONFIG_LIBDIR:/usr/share/pkgconfig"\nexport PKG_CONFIG_LIBDIR\nfi\nexec pkg-config "$@"\n' | sudo tee /usr/bin/x86_64-w64-mingw32-pkg-config
+```
+
+Create header files for WHPX API.
```
Comment (LecraM2003) [inline README.md:166]: Change "." to ":"

File: README.md [hunk 7]
```diff
@@ -113,13 +113,130 @@ pip install colorama
...
+```
+
+Build QEMU for Windows.
```
Comment (LecraM2003) [inline README.md:174]: Change "." to ":"

File: README.md [hunk 8]
```diff
@@ -113,13 +113,130 @@ pip install colorama
...
+
+```console
+mkdir build
+cd build
+../configure --cross-prefix=x86_64-w64-mingw32- --disable-guest-agent-msi --disable-werror --extra-cflags="-I/usr/x86_64-w64-mingw32/sys-root/mingw/include -I/usr/x86_64-w64-mingw32/sys-root/mingw/include/glib-2.0" --extra-ldflags="-L/usr/x86_64-w64-mingw32/sys-root/mingw/lib" --target-list=aarch64-softmmu
+make
+```
+
+Copy all the DLLs.
```
Comment (LecraM2003) [inline README.md:183]: Change "." to ":"

**Review by LecraM2003 — CHANGES_REQUESTED** [review]

**Review by LecraM2003 — APPROVED** [review]

**Comment by Filiprogrammer** [comment]
#duration: 2

**Pull request merged**